### ISSUE TITLE
Retrieve VolumeAttachments from cluster

### ISSUE THREAD

**Issue by naemono** [elastic/eck-diagnostics#109]
We likely could benefit from retrieving `VolumeAttachment`s from clusters to catch these types of errors:

```
- apiVersion: storage.k8s.io/v1
  kind: VolumeAttachment
  metadata:
    annotations:
      csi.alpha.kubernetes.io/node-id: aks-node000
    deletionGracePeriodSeconds: 0
    finalizers:
    - external-attacher/disk-csi-azure-com
    name: csi-asdf
  spec:
    attacher: disk.csi.azure.com
    nodeName: aks-node000
    source:
      persistentVolumeName: pvc-asdf
  status:
    attachError:
      message: 'rpc error: code = Unknown desc = Attach volume /subscriptions/01234/resourceGroups/rg0/providers/Microsoft.Compute/disks/pvc-asdf
        to instance aks-node000 failed with Retriable: true,
        RetryAfter: 0s, HTTPStatusCode: -1, RawError: Code="VMExtensionProvisioningTimeout"
        Message="Provisioning of VM extension AKSLinuxExtension has timed out. Extension
        provisioning has taken too long to complete. The extension did not report
        a message. More information on troubleshooting is available at https://aka.ms/vmextensionlinuxtroubleshoot"'
    attached: false
    detachError:
      message: 'rpc error: code = Internal desc = Could not detach volume /subscriptions/01234/resourceGroups/rg0/providers/Microsoft.Compute/disks/pvc-asdf
        from node aks-node0000: Retriable: true, RetryAfter: 0s,
        HTTPStatusCode: -1, RawError: Code="VMExtensionProvisioningTimeout" Message="Provisioning
        of VM extension AKSLinuxExtension has timed out. Extension provisioning has
        taken too long to complete. The extension did not report a message. More information
        on troubleshooting is available at https://aka.ms/vmextensionlinuxtroubleshoot"'
```
